mslib/mswms/mss_plot_driver.py

```diff
@@ -796,9 +797,10 @@ def _load_interpolate_timestep(self):
             cross_section = utils.interpolate_vertsec(var_data, self.lat_data, lon_data, self.lats, self.lons)
             # Create vertical interpolation factors and indices for subsequent variables
             # TODO: Improve performance for this interpolation in general
-            if name == "air_pressure":
+            if name == "air_pressure" or "air_pressure" not in self.data_vars and len(factors) == 0:
```

Comment:
I recently "automagically" added air_pressure to vertical sections in mslib/mswms/mpl_vsec.py lines 166ff. Maybe yoou could that similarly beforehand?

Reply:
I actually did use your vsec solution, but it turned out to be not quite what I needed. And instead of modifying your solution I decided to just use self.vert_data

Reply:
I'm not sure if I get this if condition right, 

factors creates the limitation that only once this is called. if we had "air_pressure" in the self.data_vars it would be on index 0. And if we don't have it while factors is empty it runs, but it can't not be executed more than once, so could that be simplified to len(factors) == 0 ?

Reply:
You are right of course.